### PR TITLE
[webview_flutter_wkwebview] Adds the `isInspectable` to `WebKitWebViewController`.

### PR DESCRIPTION
**Pull request by GwonHyeok**
Fixes: flutter/flutter#126899

Starting from iOS version 16.4, it is now possible to change the value of the isInspectable property in WKWebView.

https://developer.apple.com/documentation/webkit/wkwebview/4111163-isinspectable

Since iOS 16.4, the Web Inspector feature is disabled by default. 
Therefore, to use Web Inspector, need to change the value of isInspectable.

Here is a blog post from WebKit that discusses this
https://webkit.org/blog/13936/enabling-the-inspection-of-web-content-in-apps/

Apple Developer Forums thread (iOS 16.4 webview can not debug in safari web inspector) 
https://developer.apple.com/forums/thread/727049

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
Thanks for the contribution!

> * [x]  I listed at least one issue that this PR fixes in the description above.

Could you file an issue providing more context on this, especially since the Apple docs don't explain it? That would be helpful for reviewers, and for future understanding of this PR in context.

This will also need native unit tests that validate that it's actually calling the relevant API.

**Comment by GwonHyeok** [comment]
Since iOS 16.4 onwards, the Web Inspector feature is disabled by default. Therefore, to use Web Inspector, you need to change the value of isInspectable.

Here is a blog post from WebKit that discusses this
https://webkit.org/blog/13936/enabling-the-inspection-of-web-content-in-apps/

Apple Developer Forums thread (iOS 16.4 webview can not debug in safari web inspector) 
https://developer.apple.com/forums/thread/727049

**Comment by cyanglaz** [comment]
@GwonHyeok Thanks for the explanation! This makes sense.

Please follow the instructions in https://github.com/flutter/packages/pull/3984#issuecomment-1548433266 before we can review and land this.

I think your [comment](https://github.com/flutter/packages/pull/3984#issuecomment-1548893086) is clear enough so you can copy it over to the issue that you will create.

**Comment by GwonHyeok** [comment]
Thank you for the quick response

I have edited the PR description message

**Comment by GwonHyeok** [comment]
@cyanglaz 

I have created an issue for this PR and assigned it.

**Comment by GwonHyeok** [comment]
@cyanglaz 

I added native unit tests for this PR. 👍🏻

File: packages/webview_flutter/webview_flutter_wkwebview/pubspec.yaml
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_wkwebview/pubspec.yaml:5]: nit: Same here `3.5.0`

File: packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit/web_kit.dart
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit/web_kit.dart:1096]: According to https://webkit.org/blog/13936/enabling-the-inspection-of-web-content-in-apps/. This flag only affects each `WKWebView` separately, right?

> This decision is made for each individual WKWebView and JSContext to prevent unintentionally making it enabled for a view or context you don’t intend to be inspectable.

File: packages/webview_flutter/webview_flutter_wkwebview/lib/src/webkit_webview_controller.dart
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_wkwebview/lib/src/webkit_webview_controller.dart:534]: As stated in the other comment, I believe this flag only enables debugging for the `WebView` it is called on. This should include this.

**Review by bparrishMines — COMMENTED** [review]
Thanks for the contribution!

**Comment by GwonHyeok** [comment]
@bparrishMines 

I have reviewed all the feedback you provided and have committed the changes.

File: packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit/web_kit.dart
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit/web_kit.dart:1096]: nit: 
```suggestion
  /// Enables debugging of web contents (HTML / CSS / JavaScript) in the
  /// underlying WebView.
```

File: packages/webview_flutter/webview_flutter_wkwebview/lib/src/webkit_webview_controller.dart
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_wkwebview/lib/src/webkit_webview_controller.dart:535]:
```suggestion
  /// Whether to enable tools for debugging the current WKWebView content.
  ///
  /// It needs to be activated in each WKWebView where you want to enable it.
```

**Review by bparrishMines — APPROVED** [review]
LGTM

cc @stuartmorgan / @cyanglaz for secondary review

**Comment by GwonHyeok** [comment]
@stuartmorgan @cyanglaz 

resolved the issue caused by a conflict with the release of version 3.4.4.

**Comment by GwonHyeok** [comment]
@bparrishMines @cyanglaz @stuartmorgan 

Hello,
Why hasn't the this PR been reviewed yet?

File: packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit/web_kit.dart
Comment (cyanglaz) [inline packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit/web_kit.dart:1100]: Could you add the link to the WKWebView documentation you want to point to? Like:
[WKWebView](link)

**Review by cyanglaz — APPROVED** [review]
LGTM % nits

File: packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit/web_kit.dart
Comment (GwonHyeok) [reply packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit/web_kit.dart]: @cyanglaz 

I added a link to the document.

4629c89b12cc4a9bc6edd8ad24cfdf84217af378

**Review by cyanglaz — APPROVED** [review]
LGTM

File: packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFWebViewHostApi.m
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFWebViewHostApi.m:203]: This should also provide a `FlutterError` for unsupported versions. See https://github.com/flutter/packages/blob/main/packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFUserContentControllerHostApi.m#L55

Comment (GwonHyeok) [reply packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFWebViewHostApi.m:203]: `FWFUnsupportedVersionError` has been added.

**Review by bparrishMines — APPROVED** [review]
LGTM

**Comment by pengfeigao** [comment]
ARC Semantic Issue (Xcode): No visible @interface for 'FWFWebView' declares the selector 'setInspectable:'
/Users/xxx/.pub-cache/hosted/pub.dev/webview_flutter_wkwebview-3.6.0/ios/Classes/FWFWebViewHostApi.m:203:44

**Comment by stuartmorgan** [comment]
@pengfeigao In the future, please use the issue tracker to report issues. That's https://github.com/flutter/flutter/issues/129587

**Comment by sagarshende23** [comment]
> ARC Semantic Issue (Xcode): No visible @interface for 'FWFWebView' declares the selector 'setInspectable:'
> /Users/xxx/.pub-cache/hosted/pub.dev/webview_flutter_wkwebview-3.6.0/ios/Classes/FWFWebViewHostApi.m:203:44

**How To fix this issue I'm also getting same issue, how to solve this issue?**

Screenshot 2023-06-28 at 5 15 02 PM

**Comment by Aulig** [comment]
> > ARC Semantic Issue (Xcode): No visible @interface for 'FWFWebView' declares the selector 'setInspectable:'
> > /Users/xxx/.pub-cache/hosted/pub.dev/webview_flutter_wkwebview-3.6.0/ios/Classes/FWFWebViewHostApi.m:203:44
> 
> **How To fix this issue I'm also getting same issue, how to solve this issue?**
> 
> [1656×332 png128,7 kBScreenshot 2023-06-28 at 5 15 02 PM](https://user-images.githubusercontent.com/43273993/249451258-a1cff208-ac47-4d97-942d-ff2b8be2abbe.png)

use this workaround https://github.com/flutter/flutter/issues/129587#issuecomment-1608968526 or wait for a fix to land

**Comment by sagarshende23** [comment]
> > > ARC Semantic Issue (Xcode): No visible @interface for 'FWFWebView' declares the selector 'setInspectable:'
> > > /Users/xxx/.pub-cache/hosted/pub.dev/webview_flutter_wkwebview-3.6.0/ios/Classes/FWFWebViewHostApi.m:203:44
> > 
> > 
> > **How To fix this issue I'm also getting same issue, how to solve this issue?**
> > [1656×332 png128,7 kBScreenshot 2023-06-28 at 5 15 02 PM](https://user-images.githubusercontent.com/43273993/249451258-a1cff208-ac47-4d97-942d-ff2b8be2abbe.png)
> 
> use this workaround [flutter/flutter#129587 (comment)](https://github.com/flutter/flutter/issues/129587#issuecomment-1608968526) or wait for a fix to land

Thanks working 👍

**Comment by wubaibin** [comment]
> ARC 语义问题 (Xcode)：不可见@interface'FWFWebView' 声明选择器 'setInspectable:' /Users/xxx/.pub-cache/hosted/pub.dev/webview_flutter_wkwebview-3.6.0/ios/Classes/FWFWebViewHostApi.m:203:44

I have the same problem with you. How did you solve it

**Pull request merged**
